Tests/AsyncAlgorithmsTests/TestShare.swift

```diff
@@ -130,31 +130,31 @@ final class TestShare: XCTestCase {
     XCTAssertEqual(results2.withLock { $0 }.sorted(), [1, 2, 3, 4, 5])
   }
 
-  func test_share_with_unbounded_buffering() async {
+  func test_share_with_unbounded_buffering() async throws {
     let source = [1, 2, 3, 4, 5]
     let shared = source.async.share(bufferingPolicy: .unbounded)
 
-    let results1 = Mutex([Int]())
-    let results2 = Mutex([Int]())
+    let results1 = ManagedCriticalState([Int]())
+    let results2 = ManagedCriticalState([Int]())
     let gate1 = Gate()
     let gate2 = Gate()
 
     let consumer1 = Task {
       var iterator = shared.makeAsyncIterator()
       gate2.open()
       await gate1.enter()
-      while let value = await iterator.next(isolation: nil) {
+      while let value = await iterator.next() {
         results1.withLock { $0.append(value) }
         // Add some delay to consumer 1
-        try? await Task.sleep(for: .milliseconds(1))
+        try? await Task.sleep(nanoseconds: 1_000_000)
```

Comment:
It might be worthwhile to keep the test suite requiring a higher version for ease of deployment (and validating the newer versions folks will in time all be using) and it avoids needing to use crusty old versions of the sleep method

Reply:
@phausler I think it depends on how the general workflow for testing is set up. I think it's useful being able to run tests on legacy platforms if we support them because we found critical issues and crashes there.

I've set AsyncAlgorithms 1.1 to be compatible with iOS 14.0 because - thanks to the tests being able to run there - I discovered issues on iOS 13.0. 
I'll create a backportable local extension on `Task` to avoid the nanoseconds initializer.